src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -1598,26 +1594,52 @@
    DNS alt names, and several different fingerprint hashes of the certificate."
   [{:keys [csrdir signeddir cacert cacrl cakey]} :- CaSettings
    subject :- schema/Str]
-  (let [crl (utils/pem->ca-crl cacrl (utils/pem->ca-cert cacert cakey))]
-    (get-certificate-status* signeddir csrdir crl subject)))
+  (let [crl      (utils/pem->ca-crl cacrl (utils/pem->ca-cert cacert cakey))
+        is-cert? (fs/exists? (path-to-cert signeddir subject))]
+    (if is-cert?
+      (let [cert (utils/pem->cert (path-to-cert signeddir subject))]
+        (get-certificate-status* crl is-cert? subject cert))
+      (let [csr (utils/pem->csr (path-to-cert-request csrdir subject))]
+        (get-certificate-status* crl is-cert? subject csr)))))
+
+(schema/defn ^:always-validate get-CSRs-statuses :- [CertificateStatusResult]
+  "Get the status of only certificate requests by only going
+   through csrdir."
+  [csrdir :- schema/Str
+   crl :- CertificateRevocationList]
+  (let [pem-pattern   #"^.+\.pem$"
+        all-subjects  (map #(fs/base-name % ".pem") (fs/find-files csrdir pem-pattern))
+        all-CSRs      (map #(utils/pem->csr (path-to-cert-request csrdir %)) all-subjects)
+        is-cert?      false]
+    (map (partial get-certificate-status* crl is-cert?) all-subjects all-CSRs)))
```

Comment:
you don't need to put `is-cert?` into the `let` here, you can just use `false` directly.